File: backend/app/schemas/metric.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class MetricRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    model_id: str
    version: str
    environment: str
    timestamp: datetime
    latency_ms: float
    throughput_rpm: float
    error_rate: float
    quality_score: float
    drift_score: float
    availability: float


class MetricSummary(BaseModel):
    version: str
    environment: str
    latest: MetricRead | None
    monitoring_status: str
    last_successful_inference: datetime | None
    sample_count: int


class ModelMetricsResponse(BaseModel):
    model_id: str
    summaries: list[MetricSummary]
    history: list[MetricRead]

File: frontend/src/app/core/models/metric.ts
export type MonitoringStatus = 'HEALTHY' | 'DEGRADED' | 'UNHEALTHY';

export interface Metric {
  id: string;
  model_id: string;
  version: string;
  environment: string;
  timestamp: string;
  latency_ms: number;
  throughput_rpm: number;
  error_rate: number;
  quality_score: number;
  drift_score: number;
  availability: number;
}

export interface MetricSummary {
  version: string;
  environment: string;
  latest: Metric | null;
  monitoring_status: MonitoringStatus;
  last_successful_inference: string | null;
  sample_count: number;
}

export interface ModelMetricsResponse {
  model_id: string;
  summaries: MetricSummary[];
  history: Metric[];
}
